pandas/core/arrays/datetimes.py

```diff
@@ -680,6 +684,46 @@ def _format_native_types(
             self.asi8, tz=self.tz, format=fmt, na_rep=na_rep
         )
 
+    def _format_array(
+        self,
+        formatter: Callable | None,
+        *,
+        float_format: FloatFormatType,
+        na_rep: str = "NaN",
+        digits: int,
+        space: str | int,
+        justify: str = "right",
+        decimal: str = ".",
+        leading_space: bool | None = True,
+        quoting: int | None = None,
+    ) -> list[str]:
```

Comment:
same comment this seems like adding a lot of boilerplate that could be handled in the base class no?

Reply:
This is slightly different than the Categorical case. Categorical wants to change the `values` passed to the `fmt_klass`. This is actually changing the `fmt_klass` itself.

We *could* add some method to the interface to get the formatting class for an array. I don't really think that we want to publicly expose the ArrayFormatter interface publicly though.

If it's purely about the lines of code here, we could have a "private" `_fmt_klass` on our DatetimeArrays and check for that attribute, and use it in the base class.

```python
# in ExtensionArray._format_array
if hasattr(self, "_format_class"):
    fmt_klass = self._format_class
else:
    fmt_klass = GenericArrayFormatter
```

I dunno. This is all kind of messy.